workspace_host/interface.js

```diff
@@ -539,116 +539,125 @@ async function _getWorkspaceSettingsAsync(workspace_id) {
-          dateCache = Date.now() - 1e6;
-        docker.modem.followProgress(
-          stream,
-          (err) => {
-            if (ERR(err, callback)) return;
-            if (percentDisplayed) {
-              const toDatabase = false;
-              workspaceUtils.updateWorkspaceMessage(
-                workspace.id,
-                `Pulling image (100%)`,
-                toDatabase
-              );
-            }
-            callback(null, workspace);
-          },
-          (output) => {
-            debug('Docker pull output: ', output);
-            if ('progressDetail' in output && output.progressDetail.total) {
-              // track different states (Download/Extract)
-              // separately by making them separate keys
-              const key = `${output.id}/${output.status}`;
-              progressDetails[key] = output.progressDetail;
-            }
-            current = Object.values(progressDetails).reduce(
-              (current, detail) => detail.current + current,
-              0
-            );
-            const newTotal = Object.values(progressDetails).reduce(
-              (total, detail) => detail.total + total,
-              0
-            );
-            if (outputCount <= 200) {
-              // limit progress initially to wait for most layers to be seen
-              current = Math.min(current, (outputCount / 200) * newTotal);
-            }
-            if (newTotal > total) {
-              total = newTotal;
-              currentBase = current;
-              fractionBase = fraction;
-            }
-            if (total > 0) {
-              outputCount++;
-              const fractionIncrement =
-                total > currentBase ? (current - currentBase) / (total - currentBase) : 0;
-              fraction = fractionBase + (1 - fractionBase) * fractionIncrement;
-              const percent = Math.floor(fraction * 100);
-              const date = Date.now();
-              const percentDelta = percent - percentCache;
-              const dateDeltaSec = (date - dateCache) / 1000;
-              if (percentDelta > 0 && dateDeltaSec >= config.workspacePercentMessageRateLimitSec) {
-                percentCache = percent;
-                dateCache = date;
-                percentDisplayed = true;
-                const toDatabase = false;
-                workspaceUtils.updateWorkspaceMessage(
-                  workspace.id,
-                  `Pulling image (${percent}%)`,
-                  toDatabase
-                );
-              }
-            }
-          }
+
+        if (!percentDisplayed) {
+          resolve();
+          return;
+        }
+
+        const toDatabase = false;
+        workspaceUtils
+          .updateWorkspaceMessage(workspace.id, `Pulling image (100%)`, toDatabase)
+          .catch((err) => {
+            logger.error('Error updating workspace message', err);
+            Sentry.captureException(err);
+          })
+          .then(resolve);
```

Comment:
Should this be `.finally()`?

Reply:
I don't believe so, as the previous line is a `catch` - no errors should make its way past that. Does that sound right?

Reply:
Agreed. I don't know what I was thinking there, sorry!